hibernate-core/src/main/java/org/hibernate/dialect/DB2SqlAstTranslator.java

```diff
@@ -296,4 +290,12 @@ protected void visitReturningColumns(MutationStatement mutationStatement) {
 		// For DB2 we use #renderReturningClause to render a wrapper around the DML statement
 	}
 
+	private DatabaseVersion getDialectVersion() {
+		final var dialect = getDialect();
+		if (dialect instanceof DB2Dialect) {
+			return ((DB2Dialect) dialect).getDB2Version();
+		}
+		return dialect.getVersion();
+	}
+
```

Comment:
This approach looks a little bit uglier than the one we took for `MySQLDialect` vs `MariaDBDialect`/`TiDBDialect`.

I would prefer it if you could just follow that pattern mainly just for consistency.

(Assuming they're functionally equivalent, of course.)

Reply:
The problem for MariaDB and MySql is differently in that both dialects use a separate `AbstractSqlAstTranslator` Implementation so there are aware of the version differences between MySql and MariaDB.

Currently the `DB2SqlAstTranslator` is used as the parent for the `DB2zSqlAstTranslator` and `DB2iSqlAstTranslator`.
It might be a better alternative to set the version for `DB2SqlAstTranslator`  in the constructor and not get it from the dialect directly.
What do you think of that?

Reply:
Um, OK, that sounds fine to me, not sure what @beikov thinks of that.

Reply:
I keep this as unresolved until we have his feedback.

Reply:
>It might be a better alternative to set the version for DB2SqlAstTranslator in the constructor and not get it from the dialect directly.
>What do you think of that?

Sounds good to me.

Reply:
Alright. I will change this later this day.

Reply:
Thanks @Atgoogat, apologies for nitpicking.

Reply:
@gavinking No problem! I learned a lot and had fun :)

Reply:
@beikov I rethought this and saw that the problem might be solved exactly like in the belonging dialects by overriding the `getDB2Version()` method.

I now have changed it like that and think this is better than setting this in the constructor. Is this acceptable?

Reply:
Yeah, that's fine